### DISCUSSION TITLE
Shodan Favicon Module

### DISCUSSION THREAD

**Discussion by TheTechromancer** [blacklanternsecurity/bbot#948] [Module Requests]
@liquidsec a favicon module might be one of the pieces to building a good cloudflare bypass module. It would take in a favicon from a cloudflare-protected IP and find all the other IP addresses where the same app was deployed. Although we'd need another way to confirm whether any of them were actually the same asset.

As suggested by @amiremami

**Comment by Sh4d0wHunt3rX** [comment]
That would be great if you can also add the hash to asset inventory file.
![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/bcc7b64d-8ce0-4de0-b9e7-e45ef48167f2)

https://github.com/pielco11/fav-up
https://github.com/devanshbatham/FavFreak

Thanks 🙏

**Comment by liquidsec** [comment]
a tool to look at when considering implementing this: https://github.com/devanshbatham/FavFreak

**Comment by Sh4d0wHunt3rX** [comment]
Maybe useful: (got from orwagodfather talk. Using Shodan + Fofa

https://github.com/phor3nsic/favicon_hash_shodan

So, he searches the site on fofa first, select the similar favicons, get the hashes, and search those on shodan.

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/2c436017-7c12-4d51-83ef-be2a6127f18f)
![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/bf5f3000-b2af-4363-88b1-6a8adccb718e)

**Comment by liquidsec** [comment]
bumping as this is relevant to my current waf_bypass module. I am currently taking a chunk of the content and comparing, maybe this can be optimized by first comparing favicon
